src/analyses/goto_check.cpp

```diff
@@ -99,6 +103,12 @@ class goto_checkt
   std::unique_ptr<local_bitvector_analysist> local_bitvector_analysis;
   goto_programt::const_targett current_target;
   guard_managert guard_manager;
+  messaget log;
+
+  /// when true causes a forced collection of all allocations
+  /// (needed to allow selectively enabling checks per instruction)
+  bool do_collect_allocations;
```

Comment:
It seems like we are setting this to `true` in the constructor and no other method sets it to `false`, so wouldn't this be _always_ `true`? And if that's the case, can we completely remove this and also remove the `if(!... & !...) return;` check in `collect_allocations`?

Or did I miss a code path where it won't be `true`?

Reply:
this variable was removed and collection is now always on